ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/BitVector.hs

```diff
@@ -141,3 +158,56 @@ prop_findIthZeroInV testSetup = case testSetup of
                       $ QC.counterexample "There are not testIndex-many post-polarization-0s preceding i!"
                       $ targetsInPrecedingWords QC.=== C.getCount testIndex
                   )
+
+
+data FillInWindowSetup =
+    FillInWindowSetup
+      SomePol
+      Int -- ^ k
+      Int -- ^ size of the window to fill
+      QCGen -- ^ random generator
+      (V.Vector S) -- ^ the vector to fill
+  deriving (Show)
+
+instance QC.Arbitrary FillInWindowSetup where
+  arbitrary = do
+      pol <- QC.arbitrary
+      QC.NonNegative k <- QC.arbitrary
+      QC.NonNegative k1 <- QC.arbitrary
+      QC.NonNegative k2 <- QC.arbitrary
+      qcGen <- QC.arbitrary
+      let szV = k1 + 1
+          szW = max k szV + k2 -- k <= szW && szV <= szW
+      ss <- QC.vectorOf szV QC.arbitrary
+      pure $ FillInWindowSetup pol k szW qcGen (V.fromList ss)
+
+prop_fillInWindow :: FillInWindowSetup -> QC.Property
+prop_fillInWindow
+        (FillInWindowSetup
+          (SomePol pol)
+          k
+          szW
+          qcGen
+          v
+        ) = R.runSTGen_ qcGen $ \g -> do
+    mv <- C.unsafeThawV $ C.Vector v
+    let szV = V.length v
+    actives0 <- C.getCount <$> BV.countActivesInMV pol mv
+    c <- C.getCount <$> BV.fillInWindow pol (BV.SomeDensityWindow (C.Count k) (C.Count szW)) g mv
+    actives1 <- C.getCount <$> BV.countActivesInMV pol mv
+    pure $
+        QC.counterexample (showPol pol) $
+        QC.counterexample ("actives0 = " ++ show actives0) $
+        QC.counterexample ("actives1 = " ++ show actives1) $
+        QC.counterexample ("szV = " ++ show szV) $
+        QC.counterexample ("szW = " ++ show szW) $
+        QC.counterexample ("k = " ++ show k) $
+
+        (QC.counterexample "actives0 <= actives1" $
+          QC.property $ actives0 <= actives1)
+        QC..&&.
+        (QC.counterexample "c == actives1" $
+          QC.property $ c == actives1)
+        QC..&&.
+        (QC.counterexample "k <= actives1 + szW - szV" $
+          QC.property $ k <= actives1 + szW - szV)
```

Comment:
For better counterexample output:
```suggestion
        (QC.counterexample "actives0 <= actives1" $
          actives0 `QC.le` actives1)
        QC..&&.
        (QC.counterexample "c == actives1" $
          c QC.=== actives1)
        QC..&&.
        (QC.counterexample "k <= actives1 + szW - szV" $
          k `QC.le` actives1 + szW - szV)
```
(also requires importing from `Test.Util.QuickCheck`)

Reply:
Adopted.